cmake/IgnUtils.cmake

```diff
@@ -952,12 +952,58 @@ function(ign_create_core_library)
 
   #------------------------------------
   # Create the target for the core library, and configure it to be installed
-  _ign_add_library_or_component(
-    LIB_NAME ${PROJECT_LIBRARY_TARGET_NAME}
-    INCLUDE_DIR "${PROJECT_INCLUDE_DIR}"
-    EXPORT_BASE IGNITION_${IGN_DESIGNATION_UPPER}
-    SOURCES ${sources}
-    ${interface_option})
+
+  # Support "gz-"
+  if(${PROJECT_LIBRARY_TARGET_NAME} MATCHES "^gz-")
```

Comment:
I would create a `PROJECT_PREFIX_UPPER` variable right after we create `PROJECT_PREFIX` in [IgnConfigureProject.cmake](https://github.com/ignitionrobotics/ign-cmake/blob/main/cmake/IgnConfigureProject.cmake#L60-L65) and use that variable instead of duplicating the `MATCHES` logic

Reply:
Revert in https://github.com/ignitionrobotics/ign-cmake/commit/64bad566376334eab09bc57f77b1b6c61336e179 means we won't be using the match. :+1: